File: Difficulty: Easy/Sorted Insert Position/sorted-insert-position.java
class Solution {
    public int searchInsertK(int nums[], int k) {
        // code here
        int n = nums.length;
        int low = 0;
        int high = n-1;
        int index = n;
        
        while(low<=high) {
            int mid = (low + high) / 2;
            if(nums[mid] >= k) {
                index = mid;
                high = mid -1;
            }
            else low = mid + 1;
        }
        return index;
    }
};

File: Difficulty: Easy/Sorted Insert Position/sorted-insert-position.cpp
class Solution {
  public:
    int searchInsertK(vector<int> &nums, int target) {
        // code here
        int n = nums.size();
        int ans = n;
        
        int low = 0;
        int high = n-1;
        
        while(low<=high) {
            int mid = low + (high - low) / 2;
            
            if(nums[mid] >= target) {
                ans = mid;
                high = mid-1;
            }
            else low = mid+1;
        }
        return ans;
    }
};